src/auth/index.ts

```diff
@@ -1,117 +1,105 @@
 import axios, {AxiosStatic} from 'axios';
 import {extractAxiosErrorMessage} from '../utils';
-import {CryptoProvider, LoginDetails} from './types';
+import {CryptoProvider, LoginDetails, RegisterDetails} from './types';
 import {UserSettings} from '../shared/types/userSettings';
 import {TeamsSettings} from '../shared/types/teams';
 
 export * from './types';
 
 export class Auth {
-    private axios: AxiosStatic;
-    private readonly apiUrl: string;
-    private readonly clientName: string;
-    private readonly clientVersion: string;
+  private axios: AxiosStatic;
+  private readonly apiUrl: string;
+  private readonly clientName: string;
+  private readonly clientVersion: string;
 
-    public static client(apiUrl: string, clientName: string, clientVersion: string) {
-        return new Auth(axios, apiUrl, clientName, clientVersion);
-    }
+  public static client(apiUrl: string, clientName: string, clientVersion: string) {
+    return new Auth(axios, apiUrl, clientName, clientVersion);
+  }
 
-    constructor(axios: AxiosStatic, apiUrl: string, clientName: string, clientVersion: string) {
-        this.axios = axios;
-        this.apiUrl = apiUrl;
-        this.clientName = clientName;
-        this.clientVersion = clientVersion;
-    }
+  constructor(axios: AxiosStatic, apiUrl: string, clientName: string, clientVersion: string) {
+    this.axios = axios;
+    this.apiUrl = apiUrl;
+    this.clientName = clientName;
+    this.clientVersion = clientVersion;
+  }
 
-    public register(
-        name: string,
-        lastname: string,
-        email: string,
-        password: string,
-        mnemonic: string,
-        salt: string,
-        privateKey: string,
-        publicKey: string,
-        revocationKey: string,
-        referral?: string,
-        referrer?: string,
-    ): Promise<{
-        token: unknown,
-        user: unknown,
-        uuid: unknown
-    }> {
-        return this.axios
-            .post(`${this.apiUrl}/api/register`, {
-                name: name,
-                lastname: lastname,
-                email: email,
-                password: password,
-                mnemonic: mnemonic,
-                salt: salt,
-                privateKey: privateKey,
-                publicKey: publicKey,
-                revocationKey: revocationKey,
-                referral: referral,
-                referrer: referrer,
-            }, {
-                headers: this.headers(),
-            })
-            .then(response => {
-                return response.data;
-            })
-            .catch(error => {
-                throw new Error(extractAxiosErrorMessage(error));
-            });
-    }
+  public register(registerDetails: RegisterDetails): Promise<{
+    token: unknown,
```

Comment:
Missing types here

Reply:
@sg-gs I didn't use the same UserSettings type here on the return because the attributes that come from the server don't match exactly with out interface. There's one attribute that arrives which is not defined in the interface, and another one that is defined which doesn't comes from the server.

I'll add types for the UUID and Token, should we cast the user anyway to UserSettings?